src/plugins/discover/public/application/apps/main/components/chart/histogram.tsx

```diff
@@ -56,154 +65,186 @@ function getTimezone(uiSettings: IUiSettingsClient) {
   }
 }
 
-export class DiscoverHistogram extends Component<DiscoverHistogramProps, DiscoverHistogramState> {
-  public static propTypes = {
-    chartData: PropTypes.object,
-    timefilterUpdateHandler: PropTypes.func,
-  };
-
-  private subscription?: Subscription;
-  public state = {
-    chartsTheme: getServices().theme.chartsDefaultTheme,
-    chartsBaseTheme: getServices().theme.chartsDefaultBaseTheme,
-  };
-
-  componentDidMount() {
-    this.subscription = combineLatest([
-      getServices().theme.chartsTheme$,
-      getServices().theme.chartsBaseTheme$,
-    ]).subscribe(([chartsTheme, chartsBaseTheme]) =>
-      this.setState({ chartsTheme, chartsBaseTheme })
+export function DiscoverHistogram({
+  savedSearchData$,
+  timefilterUpdateHandler,
+  services,
+}: DiscoverHistogramProps) {
+  const [theme, setTheme] = useState({
+    chartsTheme: services.theme.chartsDefaultTheme,
+    chartsBaseTheme: services.theme.chartsDefaultBaseTheme,
+  });
+  const [state, setState] = useState<DiscoverHistogramState>({
+    chartsTheme: services.theme.chartsDefaultTheme,
+    chartsBaseTheme: services.theme.chartsDefaultBaseTheme,
+    chartData: undefined,
+    fetchStatus: FetchStatus.LOADING,
+  });
+
+  useEffect(() => {
+    const themeSubscription = combineLatest([
+      services.theme.chartsTheme$,
+      services.theme.chartsBaseTheme$,
+    ]).subscribe(([chartsTheme, chartsBaseTheme]) => {
+      if (chartsTheme !== theme.chartsTheme || chartsBaseTheme !== theme.chartsBaseTheme)
+        setTheme({ chartsTheme, chartsBaseTheme });
+    });
+    return () => {
+      return themeSubscription.unsubscribe();
+    };
```

Comment:
I've migrated the histogram component to a functional component. While doing this, I wondered, do we really need to subscribe to the chartTheme$? If this is changed by another user in background, the histogram would be displayed in dark mode, while the rest would stay while, until the browser is reloaded